### PR TITLE
Add docs for run timeouts, add MAX_RUNTIME_SECONDS_TAG to top-level API

### PR DESCRIPTION
**Pull request by dpeng817**
As the title

### CODE REVIEW

**Comment by dpeng817** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13445** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13445" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13445?utm_source=stack-comment).

File: docs/content/deployment/run-monitoring.mdx [hunk 1]
```diff
@@ -25,6 +24,27 @@ run_monitoring:
 
 When Dagster launches a run, the run stays in STARTING status until the run worker spins up and marks the run as STARTED. In the event that some failure causes the run worker to not spin up, the run might be stuck in STARTING status. The `start_timeout_seconds` offers a time limit for how long runs can hang in this state before being marked as failed.
 
+## General run timeouts
+
+After a run is marked as STARTED, it may hang indefinitely for various reasons (user API errors, network issues, etc.). `MAX_RUNTIME_SECONDS_TAG` can be used to set a timeout on a per-run basis. If the run exceeds this timeout, and run-monitoring is enabled, it will be marked as failed.
```
Comment (erinkcochran87) [inline docs/content/deployment/run-monitoring.mdx:29]: Is it `run monitoring` or `run-monitoring`? It looks like line 11 has it as `run monitoring`

File: docs/content/deployment/run-monitoring.mdx [hunk 2]
```diff
@@ -34,6 +54,9 @@ It's possible for a run worker process to crash during a run. This can happen fo
 
 If a run worker crashes, the run it's managing can hang. The monitoring daemon can run health checks on run workers for all active runs to detect this. If a failed run worker is detected (e.g. by the K8s Job having a non-zero exit code), the run is either marked as failed or resumed (see below).
 
+Note that detecting run worker crashes only works when using a run launcher other than the <PyObject object="DefaultRunLauncher" />.
```
Comment (erinkcochran87) [inline docs/content/deployment/run-monitoring.mdx:57]: Thoughts on putting this into a callout right after this section's heading? I think it'd be good to tell the user this info before they read the entire section, so they can choose to skip it if they know it doesn't apply to them.

Ex:

```
<Note>
Detecting run worker crashes only works when using a run launcher other than the <PyObject object="DefaultRunLauncher" />.
</Note>

File: docs/content/deployment/run-monitoring.mdx (hunk 1 above)
Comment (dpeng817) [reply docs/content/deployment/run-monitoring.mdx:29]: run monitoring - you're correct.

File: docs/content/deployment/run-monitoring.mdx (hunk 2 above)
Comment (dpeng817) [reply docs/content/deployment/run-monitoring.mdx:57]: good point. Adding that change for context.

**Review by erinkcochran87 — APPROVED** [review]

**Pull request merged**